### ISSUE TITLE
Offer to move VS Code to the Applications folder

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#213909]
macOS

1. Download VS Code
4. It is in the Downloads folder
5. Run it -> all good. No dialog
6. The issue is that users leave VS Code in the Downloads folder, and then they can not auto-update 🐛 (me and @digitarald saw this happen with GH folk new to VS Code)
 
I just downloaded ollama, and on first run there was a dialog to offer to move to Applications.

Can we do the same?

![Image](https://github.com/microsoft/vscode/assets/1926584/b332acff-43ed-4cdf-853e-6b3b312a2d0b)

**Comment by bpasero** [comment]
@deepak1556 isn't there support from Electron to do this? Anything that would speak against using it?

From https://github.com/electron/electron/blob/main/docs/api/app.md

![image](https://github.com/microsoft/vscode/assets/900690/4937bb4f-9d05-4d68-ba90-cd4b09c63481)

**Comment by isidorn** [comment]
@bpasero thanks for smart comments.
Yeah, we would need flag to disable, so that CI and bisect runs can circumvent this.

**Comment by deepak1556** [comment]
> isn't there support from Electron to do this? Anything that would speak against using it?

Yup, api was made specifically for this use case https://github.com/electron/electron/issues/1692. I can't think of a reason to not use it.

**Comment by isidorn** [comment]
Assigning to myself and August just to not loose track of this.

**Comment by joaomoreno** [comment]
> The issue is that users leave VS Code in the Downloads folder, and then they can not auto-update

Note that this feature would likely not solve the problem above. Not being able to auto-update is a symptom of having the app in quarantine, which macOS does by default to an app that just got downloaded. We have code to detect this: https://github.com/microsoft/vscode/blob/7cbcafcbcc88298cfdcd0238018fbbba8eb6853e/src/vs/workbench/contrib/update/browser/update.ts#L286

Before we commit to doing this, let's verify whether it helps at all with the initial problem of users being unable to update.

**Comment by salmanmkc** [comment]
Yeah, this would be useful to know. Also if there's any auto-updates, would be nice to see an alert, not just on the first time you install or run it, maybe every time, a 'don't show this again' dialog, or some other mechanism.

**Comment by orenm-gloat** [comment]
This is such a funny issue in my org, I'm helping to move to vscode as part of adoption of copilot and so many devs are not moving it to applications and can't update
So this will be a great feature to avoid this. Thanks

**Comment by mwalker-us** [comment]
Work provides VS Code (along with a number of other tools) as a portable solution (we don't install it) through an internal dev tool. For the portable solution, it would be good to disable updates since they can't be installed anyway.
